roles/ensure_machineset/tasks/derive_machineset.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Get the names of an existing worker machinesets (of any instance type)
+  command:
+    oc get machinesets -n openshift-machine-api -o
+    jsonpath='{range .items[?(@.spec.template.metadata.labels.machine\.openshift\.io/cluster-api-machine-role=="worker")]}{.metadata.name}{"\n"}{end}'
+  register: oc_get_machinesets
+  failed_when: not oc_get_machinesets.stdout
+
+- name: Create a temporary file for the original MachineSet
+  ansible.builtin.tempfile:
+    state: file
+    suffix: .machineset.temp.yaml
+  register: machineset_tempfile
```

Comment:
could you remove this task, and use
```
> {{ artifact_extra_logs_dir }}/source_machineset.json
```
instead of 
```
{{ machineset_tempfile.path }}
```
?

(` {{ artifact_extra_logs_dir }}` used to be an optional variable, but now we're heavily relying on it to store the artifacts required to precisely understand what's going on during the toolbox script execution)